system-addon/lib/TopSitesFeed.jsm

```diff
@@ -165,19 +169,22 @@ this.TopSitesFeed = class TopSitesFeed {
    * Get an image for the link preferring tippy top, rich favicon, screenshots.
    */
   async _fetchIcon(link) {
+    const hasCustomScreenshot = !!link.customScreenshotURL;
```

Comment:
We should probably short circuit first if it has a custom screenshot and do some custom behavior. Or maybe fetchIcon just shouldn't be called if we aren't using an icon anyway?

Reply:
Now this never gets called if `customScreenshotURL` is set, instead if the customScreenshot is missing it will trigger a `maybeCacheScreenshot`.